http/http-advanced/src/main/java/io/quarkus/ts/http/advanced/SseResource.java

```diff
@@ -0,0 +1,90 @@
+package io.quarkus.ts.http.advanced;
+
+import java.util.Arrays;
+import java.util.concurrent.locks.LockSupport;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.client.ClientBuilder;
+import jakarta.ws.rs.client.WebTarget;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.sse.OutboundSseEvent;
+import jakarta.ws.rs.sse.Sse;
+import jakarta.ws.rs.sse.SseEventSink;
+import jakarta.ws.rs.sse.SseEventSource;
+
+@Path("/sse")
+public class SseResource {
+    @Context
+    Sse sse;
+
+    @GET
+    @Path("/client")
+    public String sseClient() {
+        try {
+            return consumeSse();
+        }
+        // in case that https://github.com/quarkusio/quarkus/issues/36402 throws java.lang.RuntimeException: java.lang.ClassNotFoundException:
+        // catch it and return the error message
+        catch (RuntimeException exception) {
+            return exception.getMessage();
+        }
+    }
+
+    private String consumeSse() {
+        StringBuilder response = new StringBuilder();
+
+        /*
+         * Client connects to itself (to server endpoint running on same app),
+         * because for https://github.com/quarkusio/quarkus/issues/36402 to reproduce client must run on native app.
+         * Which cannot be done in test code itself.
+         * This method acts just as a client
+         */
+        WebTarget target = ClientBuilder.newClient().target("http://localhost:" + getQuarkusPort() + "/api/sse/server");
```

Comment:
`quarkus.http.host` will in production resolve to 0.0.0.0, I think it won't make difference here, but I don't fully understand why it exists otherwise. I'll let you decide.

Reply:
AFAIK setting `quarkus.http.host` to 0.0.0.0 means, that app will listen on all network interfaces. Which is not necessary here, as this app wants to connect to itself.

Reply:
I'll trust you there.